From Benjamin Mako Hill <mako@debian.org> email:
To reproduce the bug, try to output the message-id for any message in
the system using the field option to 'mu find' like:

  mu find -f i EXAMPLE

The result will not shown any message-ids but will just repeat the
letter "i". The problem seems to be that that message-id is being stored
as a Xapian term but not as a value. As a result, it can be searched
for, but not displayed.

I'm attaching a patch that solves the problem on my system. I don't know
the codebase well enough to know if I've fixed in this in "the right way."
---
 src/mu-msg-fields.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: maildir-utils.git/src/mu-msg-fields.c
===================================================================
--- maildir-utils.git.orig/src/mu-msg-fields.c	2010-10-23 16:51:22.000000000 +0900
+++ maildir-utils.git/src/mu-msg-fields.c	2010-10-23 16:51:49.000000000 +0900
@@ -130,7 +130,7 @@
 		MU_MSG_FIELD_ID_MSGID,
 		MU_MSG_FIELD_TYPE_STRING,
 		"msgid", "i", "I",  /* 'i' for Id */
-		FLAG_GMIME | FLAG_XAPIAN_TERM
+		FLAG_GMIME | FLAG_XAPIAN_TERM | FLAG_XAPIAN_VALUE
 	},
 	
 	{ 
